drivers/sht2x/sht2x.c

```diff
@@ -0,0 +1,259 @@
+
+    /* initialize the I2C bus */
+    DEBUG("[sht2x] init: acquiring I2C\n");
+    i2c_acquire(BUS);
+    DEBUG("[sht2x] init: setting master mode\n");
+    if (i2c_init_master(BUS, I2C_SPEED) < 0) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: unable to initialize I2C bus\n");
+        return SHT2X_NOI2C;
+    }
+
+    // reset the device - send a soft reset command
+    DEBUG("[sht2x] init: issuing soft reset\n");
+    result = i2c_write_byte(BUS, ADDR, SHT2X_SOFT_RESET);
+    if (result < 1) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: soft reset failed\n");
+        return SHT2X_NODEV;
+    }
+
+    // wait 15 ms for device to reset
+    DEBUG("[sht2x] init: waiting\n");
+    i2c_release(BUS);
+    xtimer_usleep(15 * 1000);
+    i2c_acquire(BUS);
+
+    // check device's responsiveness and identity by
+    // reading the user register after the reset we did
+    // and comparing with the default value which is 0x02
+    DEBUG("[sht2x] init: reading user register\n");
+    result = i2c_read_reg(BUS, ADDR, SHT2X_USER_REG_R, &user_reg);
+    if (result < 1 || (user_reg & ~SHT2X_USER_RESERVED_MASK) != 0x02) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] init - error: can't read user_reg or wrong value [0x%02x]\n", (int)user_reg);
+        return SHT2X_NODEV;
+    }
+
+    // write desired resolution to user register
+    // while keeping the rest of the register the same
+    // (only if current resolution != desired resolution)
+    DEBUG("[sht2x] init: setting resolution\n");
+    if ((user_reg & SHT2X_USER_RESOLUTION_MASK) != dev->params.resolution) {
+        result = i2c_write_reg(BUS, ADDR, SHT2X_USER_REG_W,
+            (dev->params.resolution | (user_reg & ~SHT2X_USER_RESOLUTION_MASK)));
+        if (result < 1) {
+            i2c_release(BUS);
+            DEBUG("[sht2x] init - error: user register write failed\n");
+            return SHT2X_NODEV;
+        }
+    }
+
+    DEBUG("[sht2x] init: releasing I2C\n");
+    i2c_release(BUS);
+
+    DEBUG("[sht2x] init: successful\n");
+
+    return SHT2X_OK;
+}
+
+// Returns °C (x100)
+int16_t sht2x_read_temp(sht2x_t *dev)
+{
+    uint8_t result;
+    char buffer[3];
+    
+    uint16_t st;
+    float coeff;
+    float temp;
+
+    assert(dev);
+
+    // acquire I2C bus
+    DEBUG("[sht2x] read_temp: acquiring I2C\n");
+    i2c_acquire(BUS);
+
+    // trigger measurement
+    DEBUG("[sht2x] read_temp: triggering no hold master temp read\n");
+    result = i2c_write_byte(BUS, ADDR, SHT2X_TRIG_T_MEASUREMENT_POLL);
+    if (result < 1) {
+        i2c_release(BUS);
+        DEBUG("[sht2x] read_temp - error: i2c write failed\n");
+        return SHT2X_NODEV;
+    }
+
+    // sleep for the typical time it takes to complete the measurement
+    // this depends on the resolution and is taken from the datasheet
+    switch(dev->params.resolution) {
+        case SHT2x_RES_12_14BIT:
+            xtimer_usleep(66 * 1000);
```

Comment:
as suggested above:

- remove magic numbers, and use reasonably named macros/defines instead
    - also consider naming the wait time accorind to resolution
- use `US_PER_MS` for conversion to milliseconds, not `1000`